core/common/src/main/java/alluxio/network/protocol/RPCFileReadResponse.java

```diff
@@ -85,7 +85,7 @@ public static RPCFileReadResponse decode(ByteBuf in) {
     short status = in.readShort();
 
     DataBuffer data = null;
-    if (length > 0) {
+    if (length >= 0) {
```

Comment:
I'm using `null` to indicate EOF, previously for `length == 0` we would also return null, now we just return an empty data buffer.